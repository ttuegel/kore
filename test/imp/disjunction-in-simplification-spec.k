/*
*/

module VERIFICATION
  imports IMP

  syntax Id ::= "a" [token]
  syntax Id ::= "b" [token]
  syntax Id ::= "x" [token]
  syntax AExp ::= choice (AExp, AExp) [strict]
  rule choice(A:Int, B:Int) => A #Or B [simplification, anywhere]
endmodule


module DISJUNCTION-IN-SIMPLIFICATION-SPEC
  imports VERIFICATION
  imports IMP

  rule
    <k>
      x = choice(a, b) ;
      => .K
    </k>
    <state>
      a   |-> A:Int
      b   |-> B:Int
      x |-> (_ => X:Int)
    </state>
  ensures
    X ==Int A orBool X ==Int B

endmodule
