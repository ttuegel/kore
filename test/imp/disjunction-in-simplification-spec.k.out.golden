  #True
#And
  #True
#And
  <T>
    <k>
      x = choice ( a , b );
    </k>
    <state>
      a |-> A:Int
      b |-> B:Int
      x |-> _0
    </state>
  </T>
